userspace/libscap/engine/kmod/scap_kmod.c

```diff
@@ -249,22 +297,35 @@ int32_t scap_kmod_init(scap_t *handle, scap_open_args *oargs)
 		++j;
 	}
 
-	for (int i = 0; i < SYSCALL_TABLE_SIZE; i++)
+	/* Set interesting Syscalls */
+	for(int ppm_sc = 0; ppm_sc < PPM_SC_MAX; ppm_sc++)
 	{
-		if (!handle->syscalls_of_interest[i])
+		if(oargs->ppm_sc_of_interest.ppm_sc[ppm_sc])
 		{
-			// Kmod driver event_mask check uses event_types instead of syscall nr
-			enum ppm_event_type enter_ev = g_syscall_table[i].enter_event_type;
-			enum ppm_event_type exit_ev = g_syscall_table[i].exit_event_type;
-			// Filter unmapped syscalls (that have a g_syscall_table entry with both enter_ev and exit_ev 0ed)
-			if (enter_ev != 0 && exit_ev != 0)
-			{
-				scap_unset_eventmask(handle, enter_ev);
-				scap_unset_eventmask(handle, exit_ev);
-			}
+			scap_kmod_handle_event_mask(engine, SCAP_EVENTMASK_SET, ppm_sc);
+		}
+		else
+		{
+			scap_kmod_handle_event_mask(engine, SCAP_EVENTMASK_UNSET, ppm_sc);
```

Comment:
[ultra-nit] We could refactor this to a single call to `scap_kmod_handle_event_mask`
```c
uint32_t op = oargs->ppm_sc_of_interest.ppm_sc[ppm_sc] ? SCAP_EVENTMASK_SET : SCAP_EVENTMASK_UNSET;
scap_kmod_handle_event_mask(engine, op, ppm_sc);